src/struct_mv/_hypre_struct_mv.h

```diff
@@ -2412,6 +2415,143 @@ hypre_StructMatmultCompute_core( HYPRE_Int                  nterms,
                                  hypre_Index                Mdstart,
                                  hypre_Index                Mdstride );
 
+/* struct_matmult_fuse_fff.c */
+HYPRE_Int
+hypre_StructMatmultCompute_fuse_fff( HYPRE_Int            nprod,
```

Comment:
I believe these functions will only ever be called from the file `_fuse.c` correct?  If so, I think it would be better to have a `_fuse.h` file that is explicitly included in `_fuse.c` and the other split-up files and *not* added to this general header file.

Reply:
Hi Rob, I'm not sure if it's possible to avoid having the multiprecision implementations.

I can't define them as `static` functions (and remove them from `mup.fixed`) because they need to be used in another translation unit. Maybe I'm missing something... Let me know!

Reply:
I think if you keep them in mup.fixed but take the prototypes out of `_hypre_*.h`, then no wrapper code will be written (this is how it worked before you split the files up).  You will need to add the extra include lines in each of the split up C files, though.  Let me know if I am missing something here.

Reply:
I see the point, Rob, thanks! I agree that going forward we should be more careful of what we put in `protos.h`. Just to finish this work, I moved all the remaining core prototypes to dedicate headers so we get less auto-generated mup code [028da79](https://github.com/hypre-space/hypre/pull/1393/commits/028da79ccd0e836c8ad46c7850af7449c25c8885)